forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/TppRegistrationService.java

```diff
@@ -198,44 +197,48 @@ public void verifyTPPRegistrationRequestAgainstSSA(OIDCRegistrationRequest oidcR
 
+        List<String> registrationRequestRedirectUris = registrationRequest.getRedirectUris();
 
+        // Throws if no claims available. ssaClaimRedirectUris will not be null or empty if it returns
+        Optional<List<String>> ssaClaimRedirectUris = registrationRequest.getRedirectUrisFromSSA();
         if (registrationRequestRedirectUris == null || registrationRequestRedirectUris.isEmpty() ) {
             if(!workaroundForOBDirectoryIssue){
-                throw new OBErrorException(OBRIErrorType.TPP_REGISTRATION_REQUEST_JWT_INVALID,
-                        "At least one redirect_uri must be specified");
+                String errorMessage = "No redirect_uri claims available from the Client Registration";
+                log.debug("verifyTPPRegistrationRequestAgainstSSA() {}", errorMessage);
+                throw new DynamicClientRegistrationException(errorMessage,
+                        DynamicClientRegistrationErrorType.INVALID_CLIENT_METADATA);
             } else {
-                log.debug("The redirect uris were not set in the registration request. Setting them to be the " +
-                        "same as those set in the ssa.");
-                oidcRegistrationRequest.setRedirectUris(ssaClaimRedirectUris);
+                log.debug("verifyTPPRegistrationRequestAgainstSSA() The redirect uris were not set in the " +
+                        "registration request. Setting them to be the same as those set in the ssa.");
+                if(ssaClaimRedirectUris.isPresent()) {
+                    registrationRequest.setRedirectUris(ssaClaimRedirectUris.get());
+                }
+                /*else {
+                    String errorMessage = "The Software Statement contains no redirect_uris";
+                    log.info("verifyTPPRegistrationRequestAgainstSSA() {}", errorMessage);
+                    throw new DynamicClientRegistrationException(errorMessage,
+                            DynamicClientRegistrationErrorType.INVALID_REDIRECT_URI);
+                };*/
             }
-        }
-
-        if (!ssaClaimRedirectUris.containsAll(registrationRequestRedirectUris)) {
+        } else if (ssaClaimRedirectUris.isPresent() && ssaClaimRedirectUris.get().containsAll(registrationRequestRedirectUris)) {
             log.warn("Redirect Uri in the request doesn't match the redirect URI in the SSA");
             //TODO due to OB directory bug, we won't check redirect uri for now
             if(!workaroundForOBDirectoryIssue) {
                 String errorDescription = "The redirect URI's in the registration request must be a subset of the " +
                         "URI's in the Software Statement";
-
-                throw new OIDCException(new RegistrationError()
-                        .error(RegistrationError.ErrorEnum.INVALID_REDIRECT_URI)
-                        .errorDescription(errorDescription));
+                log.info("verifyTPPRegistrationRequestAgainstSSA() {}", errorDescription);
```

Comment:
Could be better log.error instead of log.debug? because contains the error description.